### ISSUE TITLE
Spawning over flowing lava, leading to potential death loop

### ISSUE THREAD

**Issue by davepusey** [NeunEinser/bingo#210]
Minecraft 1.20.1
Fetchr 5.1.1 (originally discovered in 5.1-pre1)
Seed 1277800318

Spawning (and respawning after death) over flowing lava can potentially lead to a death loop, or starting with very low health.

See video example below.

https://youtu.be/8Mcd8QzT6GM

And snippet from Phedran's stream (Sat 26th August 2023) where we originally encountered this issue in 5.1-pre1

https://youtu.be/_l6EkyKLGL8

Managed to "fix" it during that stream by going into creative mode, flying onto the safe land beside the original spawn point, and then manually set the spawn point using /setworldspawn command.

**Comment by NeunEinser** [comment]
Thanks for the report!

In case your worldspawn is still at that location on that server, you should run `execute in overworld run setworldspawn 0 0`, otherwise this will break things in consecutive games, as respawning is not managed by setting world spawn and the map expects the spawnpoint to always be at 0 0 in the overworld dimension (which btw is a different dimension from the one games take place).

If something like this happens again, either just generate a different card, or fix it by e.g. manually covering the lava with dirt.

**Comment by davepusey** [comment]
My server has been reloaded with a newer version since that original incidient on stream, so no issues there.

This was more a report so that a proper solution/handler could be implemented into Fetchr itself.

**Comment by davepusey** [comment]
And yes, current workaround is to check for lava before starting and generating a new card if found.

**Comment by NeunEinser** [comment]
Yes, of course, this is definitely something that needs to be fixed in the game itself 😉

**Comment by NeunEinser** [comment]
And also thanks for providing the examples, videos, and the seed, very well made report 💜

**Comment by NeunEinser** [comment]
Fixed by 72ee2de33804a8e802fd520f1f8d64dfafb9a5d9.

Github should have auto closed, not sure why not.

**Comment by davepusey** [comment]
Confirmed working in 5.1.2-rc1 on 1.20.5-pre1
